### ISSUE TITLE
Enable owner/delegate to modify project remote sync

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-core#817]
Raised by @january.weiner and @holtgrewe in today's meeting.

Instead of requiring admin access, we should provide the access and an UI for the project owner/delegate to enable or revoke remote access to their project on specific target sites. This way someone could e.g. enable their project in Kiosc directly in the SODAR UI instead of requesting an admin (that's usually me) to do this.

Technically, this would be a simple thing to implement and it wouldn't necessarily even require breaking changes to the remote sync API. Just allow the owner/delegate to change the `RemoteProject` object in the db.

### New Field for RemoteSite Model

- Add `owner_modifiable` field to `RemoteSite` model
    * Boolean, `deault=True`
    * `description='Allow project owner or delegate to modify remote access'`
    * Why this? So we can easily omit test/staging sites etc from confusing users in the project update form
- Add `owner_modifiable` field to `RemoteSiteForm`
    * Only display in form if `PROJECTROLES_SITE_MODE==SITE_MODE_SOURCE`
    * **NOTE:** We don't have an API to modify these yet, so no changes needed in serializers or API views

### Management Command

- Update `addremotesite` to update this field and have a corresponding option

### Controls in ProjectForm

- In `ProjectForm`, display checkboxes for `RemoteSite` models
    * Only display and modify IF `project.type==PROJECT_TYPE_PROJECT` AND `PROJECTROLES_SITE_MODE==SITE_MODE_SOURCE`
        + Nothing should be saved regarding this if the project is a category or we are on a target site
    * Only display the `RemoteSite`s for which `owner_modifiable==True` and `mode==SITE_MODE_TARGET` and `user_display==True`
    * This should be able to be rendered using `MultipleChoiceField` and the `CheckboxSelectMultiple` widget
    * Place after `public_guest_access` before app settings
    * As legend for the checkbox selection, use "Access `get_display_name(project.type)` on remote sites"
    * Checkbox label should be `RemoteProject.name (RemoteProject.url)`
        *  If `RemoteSite.user_display==False`, add `[NOT VISIBLE TO USERS]` after label
    * Checkbox value should correspond to `RemoteProject.level` for the `RemoteProject` object corresponding to the site and the project we're editing
        + `True` IF `REMOTE_LEVEL_READ_ROLES`
        + `False` IF `REMOTE_LEVEL_NONE` OR `REMOTE_LEVEL_REVOKED` OR `RemoteProject` object for the project and site does not yet exist
        + The other levels are not supported yet so we can go with true/false checkboxes
- Implement logic for updating the values in the form
    * If set from `False` to `True`
        + If a `RemoteProject` object for site and project doesn't exist yet, create it
        + Set `RemoteProject.level` to `REMOTE_LEVEL_READ_ROLES`
        + **NOTE:** Special care should be taken when doing this when creating a project. The `Project` object must be created before we can create a `RemoteProject` object with a `ForeignKey` pointing to the project
    * If set from `True` to `False`
        + Set `RemoteProject.level` to `REMOTE_LEVEL_REVOKED`
        + Set up JQuery to add a warning label next to the site name when set from true to false: "Access to this project will be revoked"
        + Maybe even present a confirmation dialogue?
        + Only revoke if the value was true when we loaded the view!
    * If initial value is `False` and is not changed
        + Do nothing to `RemoteProject` objects. Revoking should be done only when we explicitly revoke access to a formerly enabled remote project.
- This should not be added to API views and serializers for now. We may add it later once doing the larger API update..

### REST API

(To be added via a separate issue)

### Testing

- Add field to `RemoteSite` model tests
- Add tests for `owner_modifiable` in `RemoteSiteForm`
    * Update existing tests in `TestRemoteSiteCreateView` and `TestRemoteSiteUpdateView`
    * Ensure field is not visible if we are on a target site
- Add tests for `ProjectForm`
    * Add tests to `TestProjectCreateView` and `TestProjectUpdateView`
    * Add separate tests where you create remote sites (with the new field enabled and not) and ensure the aforementioned actions succeed when enabling/disabling project access
    * Also ensure checkbox inclusion/exclusion depending on project type and site mode

### Docs

 - Update docs
    * In `app_projectroles_usage.rst` -> `Remote Projects`  -> `As Source Site`, add mention that by setting `owner_modifiable` to True the project owner/delegate can update access in the project update form.
    * We should also update screenshots, but let's do that in one shot when we are readying to release v1.0.0.

## Tasks

- [x] Add `RemoteSite.owner_modifiable` field
- [x] Add field to `RemoteSiteForm`
- [x] Add tests for `owner_modifiable` field (model and form/view)
- [x] Update `addremotesite`
- [x] Add remote access controls to `ProjectForm`
- [x] Add logic for updating `ProjectForm`
- [x] Add warning for revoking access
- [x] ~~Handle REST API (TBD)~~
- [x] Add/update tests
- [x] Update docs

**Comment by mikkonie** [comment]
Done.
